src/sv-pipeline/scripts/SplitVariants.py

```diff
@@ -0,0 +1,72 @@
+#!/bin/python
+import argparse
+
+
+def process_bed_file(input_bed, N, bca=True):
+    condition_prefixes = {
+        'gt5kb': {
+            'condition': lambda line: (line[4] == 'DEL' or line[4] == 'DUP') and (int(line[2]) - int(line[1]) >= 5000)},
+        'lt5kb': {
+            'condition': lambda line: (line[4] == 'DEL' or line[4] == 'DUP') and (int(line[2]) - int(line[1]) < 5000)},
+        'bca': {'condition': lambda line: bca and (line[4] != 'DEL' and line[4] != 'DUP' and line[4] != 'INS')},
+        'ins': {'condition': lambda line: bca and line[4] == 'INS'}
+    }
+
+    current_lines = {prefix: [] for prefix in condition_prefixes.keys()}
+    current_counts = {prefix: 0 for prefix in condition_prefixes.keys()}
+    current_suffixes = {prefix: 'a' for prefix in condition_prefixes.keys()}
+
+    with open(input_bed, 'r') as infile:
+        for line in infile:
+            line = line.strip().split('\t')
+
+            for prefix, conditions in condition_prefixes.items():
+                if conditions['condition'](line):
+                    current_lines[prefix].append('\t'.join(line))
```

Comment:
The sample_id's are included in these bed files as the 3rd column. It did not take very long to run but I don't think memory would need to be increased unless a user really wants extremely long files i.e. increases the n_per_split to a very large number

Reply:
Right, so the two main circumstances that would cause an increase in memory usage are an increase in the number of samples in the batch or an increase in N. What was the batch size you used for testing and what was the highest N? What was the memory usage for your branch and for main?

Reply:
The highest N was 20,000. When testing with various time inputs and bed files the memory for this branch fluctuated between  about a third to 3/4 that of the current main. As N increased the memory grew closer to that of the main branch but did not exceed it.